ext/RMagick/extconf.rb

```diff
@@ -343,6 +306,76 @@ def print_summary
       message summary
     end
   end
+
+  class Config
+    include Singleton
+
+    def initialize
+      @windows = (RUBY_PLATFORM =~ /mswin|mingw/)
+
+      found = find_executable('pkg-config')
+      @use_native_pkg_config = !!found
+
+      checking_for("presence of MagickWand API (ImageMagick version >= #{Magick::MIN_WAND_VERSION})") do
+        @magick_wand = Gem::Version.new(version) >= Gem::Version.new(Magick::MIN_WAND_VERSION)
+        if @magick_wand
+          Logging.message('Detected 6.9+ version, using MagickWand API')
+        else
+          Logging.message('Older version detected, using MagickCore API')
+        end
+      end
+    end
+
+    def windows?
+      @windows
+    end
+
+    def native_pkg_config?
+      @use_native_pkg_config
+    end
+
+    def with_magick_wand?
+      @magick_wand
+    end
+
+    def module_name
+      with_magick_wand? ? 'MagickWand' : 'MagickCore'
+    end
+
+    def version
+      if windows?
+        `identify -version` =~ /Version: ImageMagick (\d+\.\d+\.\d+)-+\d+ /
+        abort 'Unable to get ImageMagick version' unless Regexp.last_match(1)
```

Comment:
I feel kind of weird about using globals for getting match data. What do you think about this?

```rb
match_data = `identify -version`.match(/Version: ImageMagick (\d+\.\d+\.\d+)-+\d+ /)
abort 'Unable to get ImageMagick version' unless match_data
match_data[1]
```